Notes/Algorithms/Data Structures/Hash-Tables.c

```diff
@@ -1,37 +1,345 @@
+        return SUCCESS;
+    }
+    printf("ERROR: The key/value pair: %s and %d could not be found in the hashTable which does not match expectations. \n", key, value);
+    return ERROR;
+}
+
+// buildSearchTests
+// creates 5 test cases expected to pass the testSearch function
+// creates 5 test cases (out of range) expected to fail the testSearch function
+void buildSearchTests() {
+    printf("Build Search Test Passes: \n");
+    testSearch("KeyTwo", 20, SUCCESS);
+    testSearch("KeyEight", 80, SUCCESS);
+    testSearch("KeyThree", 30, SUCCESS);
+    testSearch("KeyFour", 40, SUCCESS);
+    testSearch("KeyNine", 90, SUCCESS);
+
+    //5 Fail
+    printf("Build Search Test Fails: \n");
+    testSearch("KeyNinety", 0, ERROR);
+    testSearch("LaLa", 10, ERROR);
+    testSearch("Error", 20, ERROR);
+    testSearch("", 30, ERROR);
+    testSearch("VERYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYYLONNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNG", 40, ERROR);
+}
+ 
+// deleteItem
+// if key/value pair is successfully deleted return SUCCESS code
+// if key/value pair cannot be deleted return ERROR code 
+int deleteItem(HashTable * hashTable, char key[], int value) {
+    int hashIndex = hashing(hashTable, key);
+    for (int i = 0; i < hashTable->size; i++) {
+        int index = (hashIndex + i) % hashTable->size;
+        if (hashTable->items[index] != NULL && hashTable->items[index]->value == value && strcmp(hashTable->items[index]->key, key) == 0) {
+            free(hashTable->items[index]->key);
+            free(hashTable->items[index]);
+            return SUCCESS;
+        } 
+    }
+    return ERROR;
+}
+
+// testDelete
+// if the key/value pair is deleted successfully, the key/value pair cannot be found in the testHash and the expectation is that this deletion is successfull a SUCCESS code 0 is returned and corresponding message printed
+// if the key/value pair is not deleted successfully, the key/value pair cannot be found in the testHash and the expectation is that this deletion fails an ERROR code -1 is returned and corresponding message printed
+int testDelete(char key[], int value, int expectedOutcome) {  
+    HashTable * testHash = createTestHashTable();  
+    int deleteSuccess = deleteItem(testHash, key, value);
+    int searchSuccess = searchHashTable(testHash, key, value);
+    if (deleteSuccess == SUCCESS && searchSuccess == ERROR && expectedOutcome == SUCCESS) {
+        printf("The key/value pair: %s and %d was deleted successfully as expected. \n", key, value);
+        return SUCCESS;
+    }  else if (deleteSuccess == ERROR && searchSuccess == ERROR && expectedOutcome == ERROR) {
+        printf("The key/value pair: %s and %d could not be deleted successfully as expected. \n", key, value);
+        return ERROR;
+    } 
+    printf("Mismatch has occurred. Result does not match expected outcome for success or failure. \n");
```

Comment:
The error message you wrote for testSearch is great.  This message could use that level of care too.

Reply:
Better commit message updated in [commit 8aa404281d0a437b762ee324b77fafa0c25c1209](https://github.com/kcarrel/Training-Notes/pull/12/commits/8aa404281d0a437b762ee324b77fafa0c25c1209)